webapp/tests/forms/steps/test_steuerlotse_step.py

```diff
@@ -0,0 +1,469 @@
+            req.session = SecureCookieSession({'form_data': create_session_form_data(self.session_data)})
+            form_step = MockFormStep(endpoint=self.endpoint_correct, next_step=next_step)
+            response = form_step.handle()
+
+            self.assertEqual(200, response.status_code)
+            # Check response data because that's where our Mock returns. Decode because response stores as bytestring
+            self.assertEqual(form_step.title, json.loads(str(response.get_data(), 'utf-8'))[0])
+
+    def test_update_session_data_is_called(self):
+        next_step = MockFinalStep
+        expected_data = {'brother': 'Luigi'}
+        with app.app_context() and app.test_request_context():
+            with patch('app.forms.steps.steuerlotse_step.override_session_data') as update_fun, \
+                    app.app_context() and app.test_request_context(
+                        path="/" + self.endpoint_correct + "/step/" + MockFormStep.name,
+                        method='GET') as req:
+                req.session = SecureCookieSession({'form_data': create_session_form_data(expected_data)})
+                form_step = MockFormStep(endpoint=self.endpoint_correct, next_step=next_step)
+                form_step.handle()
+
+                update_fun.assert_called_once_with(expected_data)
+
+    def test_yes_no_field_content_overriden_if_empty(self):
+        steps = [MockYesNoStep, MockFinalStep]
+        with app.test_request_context():
+            mock_yesno_step = MockYesNoStep(endpoint="lotse", next_step=MockRenderStep)
+        resulting_session = run_handle(mock_yesno_step, 'POST', {'yes_no_field': 'yes'})
+        resulting_session = run_handle(mock_yesno_step, 'POST', {}, resulting_session)
+        self.assertEqual({'yes_no_field': None}, deserialize_session_data(resulting_session['form_data'],
+                                                                        app.config['PERMANENT_SESSION_LIFETIME']))
+
+
+class TestSteuerlotseStep(unittest.TestCase):
+
+    def test_if_form_step_after_render_step_then_keep_data_from_older_form_step(self):
+        endpoint_correct = "lotse"
+        original_data = {'pet': 'Yoshi', 'date': '09.07.1981', 'decimal': '60.000'}
+
+        with app.app_context() and app.test_request_context():
+            first_step = MockFormWithInputStep(endpoint=endpoint_correct, next_step=MockRenderStep)
+            second_step = MockRenderStep(endpoint=endpoint_correct, next_step=MockFormStep)
+            third_step = MockFormStep(endpoint=endpoint_correct, next_step=MockFinalStep)
+
+        session = run_handle(first_step, method='POST', form_data=original_data)
+        session = run_handle(second_step, method='GET', session=session)
+        session = run_handle(third_step, method='GET', session=session)
+        self.assertTrue(set(original_data).issubset(
+            set(deserialize_session_data(session['form_data'], app.config['PERMANENT_SESSION_LIFETIME']))))
```

Comment:
Same as before, I don't think I get why we specifically test this.

Reply:
As I remember it we needed this because we might run into the problem that we do not set the session data in a display step or on get and then we might lose that data